provider/autoscaler/hcp/resource.go

```diff
@@ -134,33 +139,12 @@ func (r *ClusterAutoscalerResource) Create(ctx context.Context, request resource
 		return
 	}
 
-	autoscaler, err := r.collection.Cluster(plan.Cluster.ValueString()).Autoscaler().Get().Send()
-	if err != nil && autoscaler.Status() != http.StatusNotFound {
-		response.Diagnostics.AddError("Can't create autoscaler", fmt.Sprintf("Autoscaler for cluster '%s' might already exists. Error: %s",
-			plan.Cluster.ValueString(), err.Error()))
-		return
-	}
-
-	resource := r.collection.Cluster(plan.Cluster.ValueString())
-
-	object, err := clusterAutoscalerStateToObject(plan)
+	err = r.updateAutoscaler(ctx, plan, nil, plan.Cluster.ValueString(), r.collection)
 	if err != nil {
 		response.Diagnostics.AddError(
-			"Failed building cluster autoscaler",
+			"Failed building cluster global autoscaler state",
 			fmt.Sprintf(
-				"Failed building autoscaler for cluster '%s': %v",
-				plan.Cluster.ValueString(), err,
-			),
-		)
-		return
-	}
-
-	_, err = resource.Autoscaler().Post().Request(object).SendContext(ctx)
-	if err != nil {
-		response.Diagnostics.AddError(
-			"Failed creating cluster autoscaler",
-			fmt.Sprintf(
-				"Failed creating autoscaler for cluster '%s': %v",
+				"Failed building cluster global autoscaler state for cluster '%s': %v",
```

Comment:
Any reason to call it global?

Reply:
ehh, I've heard it before so I did it, I can remove it if you think it's unnecessary